src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2067,6 +2091,61 @@ private void LowerBoundTypeArgumentInference(NamedTypeSymbol source, NamedTypeSy
             targetTypeArguments.Free();
         }
 
+#nullable enable
+        private bool LowerBoundFunctionPointerTypeInference(TypeSymbol source, TypeSymbol target, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            if (source is not FunctionPointerTypeSymbol { Signature: { } sourceSignature } || target is not FunctionPointerTypeSymbol { Signature: { } targetSignature })
+            {
+                return false;
+            }
+
+            if (sourceSignature.ParameterCount != targetSignature.ParameterCount)
+            {
+                return false;
+            }
+
+            if (sourceSignature.RefKind != targetSignature.RefKind)
+            {
+                return false;
+            }
+
+            if (sourceSignature.ParameterRefKinds.IsDefault
+                ? !targetSignature.ParameterRefKinds.IsDefault
+                : !sourceSignature.ParameterRefKinds.SequenceEqual(targetSignature.ParameterRefKinds))
+            {
+                return false;
+            }
+
+            // Reference parameters are treated as "input" variance by default, and reference return types are treated as out variance by default.
+            // If they have a ref kind or are not reference types, then they are treated as invariant.
+            for (int i = 0; i < sourceSignature.ParameterCount; i++)
+            {
+                var sourceParam = sourceSignature.Parameters[i];
+                var targetParam = targetSignature.Parameters[i];
+
+                if ((sourceParam.Type.IsReferenceType || sourceParam.Type.IsFunctionPointer()) && sourceParam.RefKind == RefKind.None)
```

Comment:
>sourceParam.Type.IsFunctionPointer() [](start = 57, length = 36)

Does this match the specification?

Reply:
I believe I've updated the specification to match this :).

---
In reply to: [554276075](https://github.com/dotnet/roslyn/pull/50249#discussion_r554276075) [](ancestors = 554276075)